nifi-nar-bundles/nifi-hive-bundle/nifi-hive-processors/src/main/java/org/apache/nifi/processors/hive/ConvertAvroToORC.java

```diff
@@ -0,0 +1,303 @@
+                                    } catch (ArrayIndexOutOfBoundsException aioobe) {
+                                        getLogger().error("Index out of bounds at record {} for column {}, type {}, and object {}",
+                                                new Object[]{recordsInBatch, i, fieldSchema.getType().getName(), o.toString()},
+                                                aioobe);
+                                        throw new IOException(aioobe);
+                                    }
+                                }
+                            }
+                            recordCount++;
+                            recordsInBatch++;
+
+                            if (recordsInBatch == batch.getMaxSize()) {
+                                // add batch and start a new one
+                                batch.size = recordsInBatch;
+                                orcWriter.addRowBatch(batch);
+                                batch = orcSchema.createRowBatch();
+                                recordsInBatch = 0;
+                            }
+                        }
+
+                        // If there are records in the batch, add the batch
+                        if (recordsInBatch > 0) {
+                            batch.size = recordsInBatch;
+                            orcWriter.addRowBatch(batch);
+                        }
+
+                        // finished writing this record, close the writer (which will flush to the flow file)
+                        orcWriter.close();
+                        hiveAvroSchema.set(avroSchema);
+                        totalRecordCount.set(recordCount);
+                    }
+                }
+            });
+
+            final String hiveTableName = context.getProperty(HIVE_TABLE_NAME).isSet()
+                    ? context.getProperty(HIVE_TABLE_NAME).getValue()
+                    : OrcUtils.normalizeHiveTableName(hiveAvroSchema.get().getFullName());
+            String hiveDDL = OrcUtils.generateHiveDDL(hiveAvroSchema.get(), hiveTableName);
+            // Add attributes and transfer to success
+            flowFile = session.putAttribute(flowFile, RECORD_COUNT_ATTRIBUTE, Integer.toString(totalRecordCount.get()));
+            flowFile = session.putAttribute(flowFile, HIVE_DDL_ATTRIBUTE, hiveDDL);
+            StringBuilder newFilename = new StringBuilder();
+            int extensionIndex = fileName.lastIndexOf(".");
+            if (extensionIndex != -1) {
+                newFilename.append(fileName.substring(0, extensionIndex));
+            } else {
+                newFilename.append(fileName);
+            }
+            newFilename.append(".orc");
+            flowFile = session.putAttribute(flowFile, CoreAttributes.MIME_TYPE.key(), ORC_MIME_TYPE);
+            flowFile = session.putAttribute(flowFile, CoreAttributes.FILENAME.key(), newFilename.toString());
+            session.transfer(flowFile, REL_SUCCESS);
```

Comment:
I feel like this conversion is something that people may want to be able to see the performance of. We should probably explicitly emit a Provenance CONTENT_MODIFIED event that includes the amount of time that the conversion took. Would also include in the details how many records were converted.